File: operations/src/main/java/de/davelee/trams/operations/request/AdjustVehicleDelayRequest.java
package de.davelee.trams.operations.request;

import lombok.*;

/**
 * This class is part of the TraMS Operations REST API. It represents a request to adjust the delay of a vehicle in minutes.
 * @author Dave Lee
 */
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Setter
public class AdjustVehicleDelayRequest {

    //company that owns the vehicle
    private String company;

    //fleet number of the vehicle
    private String fleetNumber;

    //delay of the vehicle in minutes (a negative number indicates the delay should be reduced)
    private int delayInMinutes;

}

File: desktop/src/app/livesituation/adjustvehicledelay.request.ts
/**
 * This class defines a model for requests to adjust vehicle delay in simulation mode in TraMS.
 */
export class AdjustVehicleDelayRequest {

    private company: string;
    private fleetNumber: string;
    private delayInMinutes: number;

    /**
     * Construct a new request to adjust the vehicle delay.
     * @param company the company that we should adjust the vehicle delay for.
     * @param fleetNumber the fleet number of the vehicle that should have its delay adjusted.
     * @param delayInMinutes the delay in minutes that should be adjusted.
     */
    constructor(company: string, fleetNumber: string, delayInMinutes: number) {
        this.company = company;
        this.fleetNumber = fleetNumber;
        this.delayInMinutes = delayInMinutes;
    }
}